lambdaclass/mirra_backend
# [GH-425] [DUNGEON] Dungeon Levels config

> [!NOTE]
> Unlocks #667 

## Motivation

Add import for Dungeon Levels
Closes https://github.com/lambdaclass/afk_gacha_game/issues/425

## Summary of changes

- Add json import function
  - since no attempt cost was provided by the design doc, I just did what seemed right and added 1 Supply for every level.
  - for experience, I added some mock values in the json.
- Add json based on https://docs.google.com/spreadsheets/d/12nSgWtkOFmL_iTCDm2LmL3DW5GAgF2jW4fqfiQJRoPs/edit#gid=1226829348

## How to test it?

Run `Champions.Config.import_dungeon_levels_config()` (seeds run it automatically) and check that the data is saved succesfully.

Here are some queries for your convenience:

```
SELECT
	level_number,
	sc.name,
	l.experience_reward,
	l.max_units,
	l.attempt_cost,
	ch.name, 
	u.level
FROM public.levels l
JOIN public.campaigns c on l.campaign_id = c.id
JOIN public.super_campaigns sc on c.super_campaign_id = sc.id
JOIN public.units u on u.campaign_level_id = l.id
JOIN public.characters ch on u.character_id = ch.id
where sc.name = 'Dungeon'
ORDER BY level_number ASC 
```

```
SELECT
	level_number,
	sc.name,
	l.experience_reward,
	l.max_units,
	l.attempt_cost,
	curr.name as currency_reward_name,
	cr.amount as currency_reward_amount
FROM public.levels l
JOIN public.campaigns c on l.campaign_id = c.id
JOIN public.super_campaigns sc on c.super_campaign_id = sc.id
JOIN public.currency_rewards cr on cr.level_id = l.id
JOIN public.currencies curr on cr.currency_id = curr.id
where sc.name = 'Dungeon'
ORDER BY level_number ASC 
```

## Checklist
